base_classes/nyaml/NXopt_beam.yaml

```diff
@@ -0,0 +1,21 @@
+category: base 
+doc: |
+  A generic optical beam, us in an experimental setup for optical spectroscopy.
+
+  Characteristics of a beam, which is linked to two NXopt_element entries.
+  A "logical direction" of the beam is defined by "prev_opt_element" and "next_opt_element"
+
+  Example Names: incomming_1, incomming_2, outgoing_1, outgoing_2, input_1, output_1,
+  souce_1, mirror_1
+
+  Beam properties of the beam can be stored here: Intensity, polarization, divergence,
+  diatmeter, electric field strength, spectral intensity distribution, ...
+
+OPTICALBEAMS(NXbeam):

```

Comment:
You are not actually defining a base class here. You need to define a new class like:
```
NXoptbeam(NXbeam):
```
Note that this notation also doesn't work because we cannot extend base classes just yet (see above).

Reply:
changed as suggested and thanks for the explanation via video chat